pkg/policy/policy.go

```diff
@@ -79,6 +88,15 @@ func (c connective) Kind() string {
 	return c.kind
 }
 
+func (c connective) Selector() selector.Selector {
+	// assuming the first statement's selector is representative
+	if len(c.statements) > 0 {
+		return c.statements[0].Selector()
```

Comment:
This doesn't seem right to me...

In general I don't really have enough context in the PR description to understand why this is necessary, plus I'm not sure I understand why policy statements need to have a selector now?

Reply:
This is arguably off-spec (and very exploratory for now), but the idea is that when you pass a UCAN as a HTTP bearer (the off-spec part), you likely want to do some or all of the following:
- include "external args" (http.method, http.path, ...) into your normal invocation `args`, and have the policies operate on those elements
- evaluate the policies in multiple stages in your server because different aspects are handled in different part of the pipeline (some args might come from the HTTP header, some from the body ...)

To be clear, all this HTTP bearer features are not going to be in go-ucan, but this specific facility in this PR might come handy even outside of this scenario, for example when you want to sub-delegate on a subset of the original policies. TBD.

That HTTP bearer thing might become a UCAN extension in the future.